src/core/referencingfeaturelistmodel.h

```diff
@@ -0,0 +1,252 @@
+    QModelIndex parent( const QModelIndex &index ) const override;
+    int rowCount( const QModelIndex &parent = QModelIndex() ) const override;
+    int columnCount( const QModelIndex &parent = QModelIndex() ) const override;
+
+    QVariant data( const QModelIndex &index, int role = Qt::DisplayRole ) const override;
+
+    /**
+     * The parent feature for which this model contains the children
+     * \param feature
+     * \see feature
+     */
+    void setFeature( const QgsFeature &feature );
+
+    /**
+     * The parent feature for which this model contains the children
+     * \return the parent feature
+     * \see setFeature
+     */
+    QgsFeature feature() const;
+
+    /**
+     * The relation connectiong the parent feature with the children in this model
+     * \param relation
+     * \see relation
+     */
+    void setRelation( const QgsRelation &relation );
+
+    /**
+     * The relation connectiong the parent feature with the children in this model
+     * \return relation
+     * \see setRelation
+     */
+    QgsRelation relation() const;
+
+    /**
+     * On many-to-many relations this is the second relation connecting the children in the associationtable to their other parent
+     * \param relation The associated relation
+     * \see nmRelation
+     */
+    void setNmRelation( const QgsRelation &relation );
+
+    /**
+     * On many-to-many relations this is the second relation connecting the children in the associationtable to their other parent
+     * \return associated relation
+     * \see setNmRelation
+     */
+    QgsRelation nmRelation() const;
+
+    /**
+     * The status if the pk of the parent feature (this feature) are valid (not null)
+     * \param parentPrimariesAvailable The status if the parent pks are available
+     * \see parentPrimariesAvailable
+     */
+    void setParentPrimariesAvailable( const bool parentPrimariesAvailable );
+
+    /**
+     * The status if the pk of the parent feature (this feature) are valid (not null)
+     * It's needed to check on opening a form to add a new child
+     * \return parentPrimariesAvailable The status if the parent pks are available
+     * \see setParentPrimariesAvailable
+     */
+    bool parentPrimariesAvailable() const;
+
+    //! Reloads the model
+    Q_INVOKABLE void reload();
+    //! Deletes a feature regarding the referencing layer and the feature id \param referencingFeatureId of the selected child
+    Q_INVOKABLE void deleteFeature( QgsFeatureId referencingFeatureId );
+
+    /**
+     * Indicator if the model is currently performing any feature iteration in the background.
+     */
+    bool isLoading() const;
+
+  signals:
+    void attributeFormModelChanged();
+    void featureChanged();
+    void relationChanged();
+    void nmRelationChanged();
+    void parentPrimariesAvailableChanged();
+
+    //Indicator if the model is currently performing any feature iteration in the background.
```

Comment:
```suggestion
    //! Indicator if the model is currently performing any feature iteration in the background.
```

Reply:
removed this comment, because it's clear because of method `isLoading` with the same text in docstrings.